BaragonService/src/main/java/com/hubspot/baragon/service/BaragonServiceModule.java

```diff
@@ -381,36 +408,65 @@ public AmazonElasticLoadBalancingClient providesAwsElbClientV1(Optional<ElbConfi
         .build());
   }
 
+  @Provides
+  public AWSStaticCredentialsProvider providesAwsStaticCredentialsProvider(Optional<ElbConfiguration> configuration) {
+    return new AWSStaticCredentialsProvider(
+        new BasicAWSCredentials(configuration.get().getAwsAccessKeyId(),
+            configuration.get().getAwsAccessKeySecret())
+    );
+  }
+
+  @Provides
```

Comment:
Curious why these are broken out into separate injectable components? We're not taking advantage of this in any tests AFACIT.

Also none of these are singletons, most concerningly the ELB client.

Reply:
:doh: not singletons, that's a massive oversight on my part. Mostly broken into separate components so they can be reused by the v1 and v2 clients